tasks/eva_2429/accession_incremental_migration.py

```diff
@@ -0,0 +1,75 @@
+import os.path
+
+import psycopg2
+import psycopg2.extras
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query
+
+from migration_util import export_data
+from migration_util import invalidate_and_set_db
+from migration_util import write_query_to_file
+
+logger = logging_config.get_logger(__name__)
+
+project_key = "projectAccession"
+assembly_key = "assemblyAccession"
+accession_query_file_name = "accession_query.txt"
+accession_db = "eva_accession_sharded"
+accession_collection = "submittedVariantEntity"
+pgsql_accession_connection_uri = "postgresql://pgsql-hxvm7-010.ebi.ac.uk:5432/vrnvaraccjtpro"
+pgsql_accession_user = "evaaccjt"
+
+
+def find_accession_studies_eligible_for_migration(migration_start_time, migration_end_time):
+    with psycopg2.connect(pgsql_accession_connection_uri, user=pgsql_accession_user) as metadata_connection_handle:
+        query_string = f"select bjep.job_execution_id, bjep.key_name, bjep.string_val, bje.start_time \
+                     from batch_job_execution bje join batch_job_execution_params bjep \
+                     on bje.job_execution_id=bjep.job_execution_id \
+                     where bjep.key_name in ('{project_key}', '{assembly_key}')  \
+                     and bje.start_time > '{migration_start_time}' and bje.start_time < '{migration_end_time}' \
+                     order by bjep.job_execution_id desc , bjep.key_name"
+
+    query_result = get_all_results_for_query(metadata_connection_handle, query_string)
+    logger.info(f"\nStudies eligible for migration : {query_result}")
+
+    job_parameter_combine = {}
+    for job_id, key_name, key_value, start_time in query_result:
+        job_parameter_combine.setdefault(job_id, {}).update({key_name: key_value})
```

Comment:
Yes, we can use defaultdict for this, but defaultdict will return a list of dictionaries for each entry and not a single dictionary.  
Due to the format we want the final data to be in (a set of tuples of study and seq, where study is the first entry in each tuple), those will have to be again iterated and converted into a single dict in the next loop, so that I can create the tuple in exact order using key_name.

Reply:
updated